pennylane/template.py

```diff
@@ -0,0 +1,225 @@
+    A circuit suitable for usage in a circuit-centric classifier.
+
+    Constructs the circuit of a circuit-centric quantum classifier :cite:`schuld2018circuit`
+    with ``len(weights)`` blocks on the given wires with the provided weights.
+    Each element of weights must be a an array of size ``len(wires)*3``.
+
+    Args:
+        weights (array[float]): shape ``(len(weights), len(wires), 3)`` array of weights
+        periodic (bool): whether to use periodic boundary conditions when
+                         applying imprimitive gates
+        ranges (Sequence[int]): Ranges of the imprimitive gates in the
+                                respective blocks
+        imprimitive_gate (pennylane.ops.Operation): Imprimitive gate to use, defaults to :class:`~.CNOT`
+        wires (Sequence[int]): Wires the circuit-centric classifier circuit should act on
+    """
+    if ranges is None:
+        ranges = [1]*len(weights)
+    for block_weights, block_range in zip(weights, ranges):
+        CircuitCentricBlock(block_weights, r=block_range, periodic=periodic, imprimitive_gate=imprimitive_gate, wires=wires)
+
+
+def CircuitCentricBlock(weights, periodic=True, r=1, imprimitive_gate=CNOT, wires=None):
+    """pennylane.template.CircuitCentricBlock(weights, periodic=True, r=1, imprimitive_gate=qml.CNOT, wires)
+    An individual block of a circuit-centric classifier circuit.
+
+    Args:
+        weights (array[float]): shape ``(len(wires), 3)`` array of weights
+        periodic (bool): Whether to use periodic boundary conditions when
+                         applying imprimitive gates
+        r (Sequence[int]): Range of the imprimitive gates of this block
+        imprimitive_gate (pennylane.ops.Operation): Imprimitive gate to use, defaults to :class:`~.CNOT`
+        wires (Sequence[int]): Wires the block should act on
+    """
+    for i, wire in enumerate(wires):
+        Rot(weights[i, 0], weights[i, 1], weights[i, 2], wires=wire)
+
+    num_wires = len(wires)
+    for i in range(num_wires) if periodic else range(num_wires-1):
+        imprimitive_gate(wires=[wires[i], wires[(i+r) % num_wires]])
+
+
+def CVNeuralNet(weights, wires=None):
+    """pennylane.template.CVNeuralNet(weights, wires)
+    A CV Quantum Neural Network
+
+    Implements the CV Quantum Neural Network (CVQNN) architecture from
+    :cite:`killoran2018continuous` for an arbitrary number of wires
+    and layers.
+
+    See :func:`CVNeuralNetLayer` for details of the expected format of
+    input parameters.
+
+    Args:
+        weights (array[array]): Array of weights for each layer of the CV
+                                neural network
+        wires (Sequence[int]): Wires the CVQNN should act on
+    """
+    for layer_weights in weights:
+        CVNeuralNetLayer(*layer_weights, wires=wires)
+
+
+def CVNeuralNetLayer(theta1, phi1, s, theta2, phi2, r, k, wires=None): #pylint: disable-msg=too-many-arguments
```

Comment:
The general CVQNN architecture has complex-valued displacement (only the embedding of a classical NN needs real-valued displacement).
Let's use `r` for squeezing magnitude, then `a` and `phi_a` for displacement

Reply:
implemented in c87ffc73d4f564e69626e7ff6ab9f33664c2b924
Do you also want squeezing angles `phi_r`?

Reply:
Good question. I suppose yes, to be consistent with Dgate